lua/ps2/server/sv_pointshopcontroller.lua

```diff
@@ -783,24 +783,28 @@ local function calculateRefundAmounts( itemsToRefund )
     return refundsByPlayer
 end
 
-local function removeSingleItem( itemClass, refund )
+local function removeSingleItem( itemClass, refund, refundCurrentPrice )
     return Pointshop2.DB.DoQuery( Format( [[
         SELECT item.id, item.data, COALESCE(inventories.ownerId, ps2_equipmentslot.ownerId) AS ownerId, ps2_equipmentslot.slotName
         FROM kinv_items AS item
         LEFT JOIN inventories ON inventories.id = item.inventory_id
         LEFT JOIN ps2_equipmentslot ON ps2_equipmentslot.itemId = item.id
         WHERE item.itemClass = %s
-    ]], Pointshop2.DB.SQLStr( itemClass ) ) )
+    ]], Pointshop2.DB.SQLStr( itemClass.name ) ) )
         :Then( function( results )
             results = results or {}
             -- Deserialize purchaseData (this is usually done in LibK but we're querying manually)
             results = LibK._.map( results, function( row )
-                row.purchaseData = util.JSONToTable( row.data ).purchaseData
+                return row.purchaseData = util.JSONToTable( row.data ).purchaseData
             end )
 
             local refundPromise = Promise.Resolve()
             if refund then
-                local refundsByPlayer = calculateRefundAmounts( refund )
+                local currentPrice = false
+				if refundCurrentPrice then
+					currentPrice = table.Inherit(itemClass.static.Price, { points = 0, premiumPoints = 0 })
+				end
+                local refundsByPlayer = calculateRefundAmounts( refund, currentPrice )
```

Comment:
The issue is the itemClass does not get passed into calculateRefundAmounts (At least from what I found using MsgAll and stuff, allitle hard to debug the tables when PrintTable does nothing.)

If you can get the itemClass in it, then that would be a good idea.

Reply:
If you're up for giving that a try I think it'd make sense - in the diff here I can only see that `currentPrice` is passed in to `calculateRefundAmounts` but it's not used. Am I missing something here it looks like refunding the current price shouldn't be able to work because the value doesn't make it all the way into the calculation 😄

Reply:
Seems I copied the wrong revision of my code when making this PR. Didn't want to push the whole file I developed this in since Dinks runs on an older version of PS2. (Due to some weird bug that also involves PAC3, not sure since I haven't seen it myself)    I have pushed the info now. Going to re-do my testing again tho. Just to be sure it all works properly.